modules/api/src/main/java/net/countercraft/movecraft/craft/Craft.java

```diff
@@ -433,6 +427,93 @@ public int getWaterLine(){
         return waterLine;
     }
 
+    public boolean shouldSink() {
+        int totalNonNegligibleBlocks = 0;
+        int totalNonNegligibleWaterBlocks = 0;
+        HashMap<List<Integer>, Integer> foundFlyBlocks = new HashMap<>();
+        HashMap<List<Integer>, Integer> foundMoveBlocks = new HashMap<>();
+        // go through each block in the blocklist, and
+        // if its in the FlyBlocks, total up the number
+        // of them
+        for (MovecraftLocation l : hitBox) {
+            int blockID = w.getBlockAt(l.getX(), l.getY(), l.getZ()).getTypeId();
+            int dataID = (int) w.getBlockAt(l.getX(), l.getY(), l.getZ()).getData();
+            int shiftedID = (blockID << 4) + dataID + 10000;
+            for (List<Integer> flyBlockDef : type.getFlyBlocks().keySet()) {
+                if (flyBlockDef.contains(blockID) || flyBlockDef.contains(shiftedID)) {
+                    foundFlyBlocks.merge(flyBlockDef, 1, (a, b) -> a + b);
+                }
+            }
+            for (List<Integer> moveBlockDef : type.getMoveBlocks().keySet()) {
+                if (moveBlockDef.contains(blockID) || moveBlockDef.contains(shiftedID)) {
+                    foundMoveBlocks.merge(moveBlockDef, 1, (a, b) -> a + b);
+                }
+            }
+
+            if (blockID != 0 && blockID != 51) {
+                totalNonNegligibleBlocks++;
+            }
+            if (blockID != 0 && blockID != 51 && blockID != 8 && blockID != 9) {
+                totalNonNegligibleWaterBlocks++;
+            }
+        }
+
+        // now see if any of the resulting percentagesit
+        // are below the threshold specified in
+        // SinkPercent
+
+        for (List<Integer> i : type.getFlyBlocks().keySet()) {
+            int numfound = 0;
+            if (foundFlyBlocks.get(i) != null) {
+                numfound = foundFlyBlocks.get(i);
+            }
+            double percent = ((double) numfound / (double) totalNonNegligibleBlocks) * 100.0;
+            double flyPercent = type.getFlyBlocks().get(i).get(0);
+            double sinkPercent = flyPercent * type.getSinkPercent() / 100.0;
+            if (percent < sinkPercent) {
+                return true;

```

Comment:
I believe this is a change in behavior?

Reply:
it's copy-pasted from the original code, so i don't believe it is. What exactly would this change?

Reply:
the code here returns, the original code sets a boolean. Returning causes the function to cease execution and thus any function calls after this will not happen, such as the sound being sent to players later in the method

Reply:
ah, alright